Lib/test/test_codecs.py

```diff
@@ -1154,6 +1154,19 @@ def test_stream_bare(self):
             got = ostream.getvalue()
             self.assertEqual(got, unistring)
 
+class FileClosesOnRaiseTest(unittest.TestCase):
```

Comment:
I think there is no need to add a new class for this test. You can add it in `CodecsModuleTest`.